app/forms/questionnaire_form.py

```diff
@@ -253,33 +253,34 @@ def get_data(self, answer_id):
         attr = getattr(self, answer_id)
         return attr.raw_data[0] if attr.raw_data else ''
 
-    def option_has_other(self, answer_id, option_index):
-        if not self.options_with_children:
-            self.options_with_children = self.schema.get_parent_options_for_block(self.block_json['id'])
+    def get_detail_answer(self, answer_id, option_index):
+        if not self.options_with_detail_answer:
+            self.options_with_detail_answer = self.schema.get_options_with_detail_answers_for_block(self.block_json['id'])
 
-        if answer_id in self.options_with_children and self.options_with_children[answer_id]['index'] == option_index:
-            return True
-        return False
-
-    def get_other_answer(self, answer_id, option_index):
-        if not self.options_with_children:
-            self.options_with_children = self.schema.get_parent_options_for_block(self.block_json['id'])
-
-        if answer_id in self.options_with_children and self.options_with_children[answer_id]['index'] == option_index:
-            return getattr(self, self.options_with_children[answer_id]['child_answer_id'])
+        for option in self.options_with_detail_answer.get(answer_id, []):
+            if option['index'] == option_index:
+                return getattr(self, option['detail_answer_id'])
         return None
 
 
 def get_answer_fields(question, data, error_messages, schema, answer_store, metadata, group_instance, group_instance_id):
     answer_fields = {}
     for answer in question.get('answers', []):
-        if 'parent_answer_id' in answer and answer['parent_answer_id'] in data and \
-                data[answer['parent_answer_id']] == 'Other':
-            answer['mandatory'] = \
-                next(a['mandatory'] for a in question['answers'] if a['id'] == answer['parent_answer_id'])
+
+        for option in answer.get('options', []):
+            if 'detail_answer' in option:
+                detail_answer = option['detail_answer']
+                detail_answer_error_messages = detail_answer['validation']['messages'] if detail_answer.get('validation') else error_messages
+                if option['value'] not in data.values():
+                    detail_answer['mandatory'] = False
```

Comment:
I'll try and find an alternative as I don't like this bit of code anyway. It's a simplification of what already existed. 

This code deals with the situation whereby if the additional text field associated with an option is mandatory it would check validation regardless of if that option was selected or not. This overrides that to say only validate its mandatory status if the associated option is chosen.

Reply:
Solution is to pass in a boolean when creating the field to disable validation.